test/cppsuite/test_harness/workload_generator.h

```diff
@@ -0,0 +1,166 @@
+/* Include guard. */
+#ifndef WORKLOAD_GENERATOR_H
+#define WORKLOAD_GENERATOR_H
+
+#include <cstdint>
+#include <map>
+#include <vector>
+
+extern "C" {
+#include "test_util.h"
+}
+
+#include "api_const.h"
+#include "configuration_settings.h"
+#include "random_generator.h"
+
+#define DEBUG_ERROR 0
+#define DEBUG_INFO 1
+
+namespace test_harness {
+class workload_generator {
+    public:
+    workload_generator(test_harness::configuration *configuration)
+    {
+        _configuration = configuration;
+    }
+
+    ~workload_generator()
+    {
+        if (_session != nullptr) {
+            if (_session->close(_session, NULL) != 0)
+                /* Failing to close session is not blocking. */
+                debug_info("Failed to close session, shutting down uncleanly", DEBUG_ERROR);
+            _session = nullptr;
+        }
+
+        if (_conn != nullptr) {
+            if (_conn->close(_conn, NULL) != 0)
+                /* Failing to close connection is not blocking. */
+                debug_info("Failed to close connection, shutting down uncleanly", DEBUG_ERROR);
+            _conn = nullptr;
+        }
+    }
+
+    /*
+     * Function that performs the following steps using the configuration that is defined by the
+     * test:
+     *  - Create the working dir.
+     *  - Open a connection.
+     *  - Open a session.
+     *  - Create n collections as per the configuration.
+     *      - Open a cursor on each collection.
+     *      - Insert m key/value pairs in each collection. Values are random strings which size is
+     * defined by the configuration.
+     */
+    int
+    load(const char *home = DEFAULT_DIR)
+    {
+        WT_CURSOR *cursor;
+        int64_t collection_count, key_count, value_size;
+        std::string collection_name;
+
+        cursor = nullptr;
+        collection_count = key_count = value_size = 0;
+        collection_name = "";
+
+        /* Create the working dir. */
+        testutil_make_work_dir(home);
+
+        /* Open connection. */
+        WT_RET(wiredtiger_open(home, NULL, test_harness::CONNECTION_CREATE, &_conn));
```

Comment:
I don't think we should call wiredtiger open here in the workload generator. I think we should do that outside in the higher level to be able to open wiredtiger with different configs. At least can we pass the open config into this function?

I'm totally fine if this is a temporary solution.

Reply:
Open config will likely be part of the configuration_options passed in, until we have a need to pass open config to `wiredtiger_open` I think we can keep it like this.

Reply:
So yeah this is a temporary solution.